src/axom/primal/operators/closest_point.hpp

```diff
@@ -23,6 +24,50 @@ namespace axom
 {
 namespace primal
 {
+
+/*!
+ * \brief Computes the closest point from a point, P, to a given segment.
+ *
+ * \param [in] P the query point
+ * \param [in] seg user-supplied segment
+ * \param [in] EPS fuzz factor for equality comparisons
+ * \return cp the closest point from a point P and a segment
+ */
+template <typename T, int NDIMS>
+AXOM_HOST_DEVICE inline Point<T, NDIMS> closest_point(const Point<T, NDIMS>& P,
+                                                      const Segment<T, NDIMS>& seg,
+                                                      double EPS = 1E-8) {
+  using PointType = Point<T, NDIMS>;
+  using VectorType = Vector<T, NDIMS>;
+
+  using detail::isGeq;
+  using detail::isLeq;
+
+  const PointType& A = seg[0];
+  const PointType& B = seg[1];
+
+  const VectorType AB(A, B);
+  const T squaredNormAB = AB.squared_norm();
+
```

Comment:
Suggestion (minor): 

It might be slightly clearer to use
```cpp
constexpr T ZERO {0.};
constexpr T ONE {1.};
```
for the comparisons to zero and one

or perhaps to change `T(0}` to either `T {0.}` or `static_cast<T>(0.)`

Reply:
Good idea. Done.